h/views/account_signup.py

```diff
@@ -191,19 +220,22 @@ def validation_failure(self):
         # When reloading the page replace the idinfo token with a fresh one.
         form_data.update(
             encode_idinfo_token(
-                self.jwt_service, orcid_id, JWTIssuers.SIGNUP_VALIDATION_FAILURE_ORCID
+                self.jwt_service,
+                provider_unique_id,
+                self.settings.issuer,
+                self.settings.audience,
```

Comment:
Audience was hard-coded in the `encode_idinfo_token()` method but we actually need it to vary per-provider, so added a new `audience` argument here.